File: thecube/python/Thecube.py
class Thecube():

    def addDigits(self, text):
        """Returns the digit's sum of given text"""
        result = 0
        for pos in range(0, len(text)):
            if text[pos].isdigit():
                result = result + int(text[pos])
        return result

    def isPrimeNumber(self, number):
        """Determines if a number is prime"""
        for previousNumber in range(number - 1, 1, -1):
            if number % previousNumber == 0:
                return False
        return True

    def containsPrimeNumber(self, code1, code2, code3):
        """The Cube kata"""
        if (self.isPrimeNumber(self.addDigits(code1))):
            return True
        if (self.isPrimeNumber(self.addDigits(code2))):
            return True
        if (self.isPrimeNumber(self.addDigits(code3))):
            return True
        return False

File: thecube/java/src/main/java/com/sourcerebels/thecube/TheCube.java
package com.sourcerebels.thecube;

import static org.apache.commons.lang.CharUtils.*;

public class TheCube {

	public boolean containsPrimeNumber(String code1, String code2, String code3) {

		return isPrimeNumber(sumDigits(code1))
				|| isPrimeNumber(sumDigits(code2))
				|| isPrimeNumber(sumDigits(code3));
	}

	private boolean isPrimeNumber(int number) {

		boolean result = false;
		if (!isZeroOrOne(number)) {
			for (int i = number - 1; i > 1; i--) {
				if (number % i == 0) {
					return false;
				}
			}
			result = true;
		}
		return result;
	}

	private boolean isZeroOrOne(int number) {
		return number == 0 || number == 1;
	}

	private int sumDigits(String code) {

		int result = 0;
		for (int i = 0; i < code.length(); i++) {
			result += toIntValue(code.charAt(i));
		}
		return result;
	}
}
